### PR TITLE
Hil driver

### PR DESCRIPTION
**Pull request by kyle-hughes21**
I know that merging this manually will take some doing, but I just wanted to throw this up here.  I have several questions for Dan regarding the update methods.

### CODE REVIEW

**Comment by SahilTikale** [comment]
Make a mock driver with the all the "mocking" all the functionality that a HIL driver is supposed to have. 
This way you can make sure that the interface class you ensures the interfaces you require for any network isolation driver like HIL. Also having the mock driver makes it easy to test the functionality on your own system.

**Comment by SahilTikale** [comment]
what is the difference between `hilDriver` and `hil_driver`. Why do have two drivers doing almost similar things ?

**Comment by SahilTikale** [comment]
Once you write the mock driver, try to write some unit tests showing how the driver is used by QUADS to call HIL and HIL like systems.

**Comment by kyle-hughes21** [comment]
Sahil, this driver is not quite finished, as you point out, and hil_driver is supposed to be deleted in favor of an updated HilDriver(just for naming convention purposes, I've fixed this in my local files).  Isaac and I have discussed some necessary further changes, and I will definitely take your advice about writing a few unit tests.

File: lib/libquads.py [hunk 1]
```diff
@@ -82,7 +84,8 @@ def __init__(self, config, statedir, movecommand, datearg, syncstate, initialize
         self.datearg = datearg
         self.logger = logging.getLogger("quads.Quads")
         self.logger.setLevel(logging.DEBUG)
-        set_hardware_service(JuniperDriver())
+        #set_hardware_service(JuniperDriver())  Taking away the hardcoding for Juniper
+        set_hardware_service(HilDriver()) #temporary hardcode to HIL
```
Comment (portante) [inline lib/libquads.py:88]: Can we remove this replaced-hardcoded driver and do this from the configuration file?

File: lib/test_mock.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
 #!/bin/python
 import sys
 from hardware_services.hardware_service import get_hardware_service, set_hardware_service
-from hardware_services.hardware_drivers.mock_driver import MockDriver
+from hardware_services.hardware_drivers.hil_driver import HilDriver
```
Comment (portante) [inline lib/test_mock.py:4]: Why are we doing this in the test environment?  Wouldn't we always want to use Mock here?

File: lib/test_mock.py [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 
 cls = get_hardware_service()
 
-cls.update_cloud()
-cls.update_host()
-cls.remove_cloud()
-cls.remove_host()
-cls.move_hosts()
-cls.list_clouds()
+#cls.update_cloud()
```
Comment (portante) [inline lib/test_mock.py:10]: This seems bad ...

**Review by portante — CHANGES_REQUESTED** [review]

**Comment by portante** [comment]
@Hugh472, how does this PR reconcile with #30?

**Comment by saacg** [comment]
closing as it has been superseded by more recent work.

**Pull request closed without merging**